lib/features/artifacts.js

```diff
@@ -128,7 +128,22 @@ export default class Artifacts {
       };
 
       try {
-        await uploadArtifact(taskHandler, stream, entryName, expiry, headers);
+        await new Promise((accept, reject) => {
+          let numRetries = taskHandler.runtime.retry.numRetries;
+          let retry = () => {
+            uploadArtifact(taskHandler, stream, entryName, expiry, headers)
+              .then(accept)
+              .catch((err) => {
+                if (numRetries--) {
+                  taskHandler.runtime.log('retrying artifact upload');
+                  setTimeout(retry, taskHandler.runtime.retry.interval);
```

Comment:
/me notes the https://www.npmjs.com/package/promise-retry and https://www.npmjs.com/package/retry packages, both of which have configurable backoff and jitter